tests/break-calls/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -116,3 +116,50 @@ runtimeAgent.getProperties(
 
 ================================================================================
 `;
+
+exports[`react.js 1`] = `
+====================================options=====================================
+parsers: ["flow", "typescript"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+function helloWorld() {
+  useEffect(() => {
+    // do something
+  }, [props.value])
+}
+
+function helloWorld() {
+  useEffect(() => {
+    // do something
+  }, [props.value, props.value, props.value, props.value, props.value, props.value, props.value, props.value, props.value, props.value, props.value])
+}
+
+
+=====================================output=====================================
+function helloWorld() {
+  useEffect(() => {
+    // do something
+  }, [props.value]);
+}
+
+function helloWorld() {
+  useEffect(() => {
+    // do something
+  }, [
+    props.value,
+    props.value,
```

Comment:
I'm not sure if we should hug in this case.

cc @prettier/core @vjeux

Reply:
I think it makes sense to hug here, especially since this array is usually short.

Reply:
This currently applies to all call expressions (not `useXXX`-only) so it could be long, and also we did not hug for objects:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAFnADYkQDqEATiQCYAUAlPsADpT77oDOcAoqlRxsDZvgC8APlYcuXAPQL8dCPh4QAtnBiEcUAOZz8AXwA0rfAEMbtu1Yv2njp3Zeub7j19emmHExAzEAgAB1xoHmRQK2pqCAB3AAVYhCiUKwA3CBw6IJAAI2orMABrHQBlUJL9A2QYanQ4YMIYTRIKPXgearA4CrScXEyhgE9kcB4o4P0+ahgk4oNNK2RUKxI+YIArHgAPACFissqrbQAZfTg1ja2QXb2K2pI4AEV0CHgbzeaQauo5hMClYCqNyFB8qFqPoYBRcrpkAAOAAMwShED4FGKoQmULgc0y12C1DgAEd0DgSYsrMtVkh1j9gnxNDh6o1fjxnm8Pl96bdfjAQXC6AikAAmYINKw4Ei1ADCWhWEyg0CJIF4cAAKiD0gy+CYTEA)
```sh
--parser babylon
```

**Input:**
```jsx
function helloWorld() {
  useEffect(() => {
    // do something
  }, { aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa })
}
```

**Output:**
```jsx
function helloWorld() {
  useEffect(
    () => {
      // do something
    },
    {
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa
    }
  );
}

```

Reply:
Current output:

**Prettier pr-5608**
[Playground link](https://deploy-preview-5608--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAFnADYkQDqEATiQCYAUAlPsADpT77oDOcAoqlRxsDZvgC8APlYcuXAPQL8dCPh4QAtnBiEcUAOZz8AXwA0rfAEMbtu1Yv2njp3Zeub7j19emmHExAzEAgAB1xoHmRQK2pqCAB3AAVYhCiUKwA3CBw6IJAAI2orMABrHQBlUJL9A2QYanQ4YMIYTRIKPXgearA4CrScXEyhgE9kcB4o4P0+ahgk4oNNK2RUKxI+YIArHgAPACFissqrbQAZfTg1ja2QXb2K2pI4AEV0CHgbzeaQauo5hMClYCqNyFB8qFqPoYBRcrpkAAOAAMwShED4FGKoQmULgc0y12C1DgAEd0DgSYsrMtVkh1j9gnxNDh6o1fjxnm8Pl96bdfjAQXC6AikAAmYINKw4Ei1ADCWhWEyg0CJIF4cAAKiD0gy+CYTEA)
```sh
--parser babylon
```

**Input:**
```jsx
function helloWorld() {
  useEffect(() => {
    // do something
  }, { aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa, aaaaaaaaaaa })
}
```

**Output:**
```jsx
function helloWorld() {
  useEffect(
    () => {
      // do something
    },
    {
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa
    }
  );
}

```

Reply:
I don’t really mind either way personally.

Reply:
I know I'm late to the party but I think we should not hug... this feels inconsistent to me.

```js
function helloWorld() {
  useEffect(
    () => {
      // do something
    },
    {
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa,
      aaaaaaaaaaa
    }
  );
}

function helloWorld() {
  useEffect(() => {
    // do something
  }, [
    aaaaaaaaaaa,
    aaaaaaaaaaa,
    aaaaaaaaaaa,
    aaaaaaaaaaa,
    aaaaaaaaaaa,
    aaaaaaaaaaa
  ]);
}
```

Reply:
Since this will be used more often now that React hooks are going to be released, I feel that we should hug the arrays to avoid excessive indentation (like we do with `describe` and `it`). Should I open a PR to hug objects too?

Reply:
> Since this will be used more often now that React hooks are going to be released

This is not specific to React and we can't possibly know how common it is. I don't see much benefit with the new formatting to justify the churn, as I don't think the indentation is excessive. (I personally prefer without hugging since it's clear and when hugging it's too cluttered, but that's my opinion and it's besides the point. My point is specific to churn without clear benefits)

To be clear, I'm only talking about the case when the array is too long and it breaks. For small arrays, I think this change is good.

Reply:
This specific implementation wouldn’t be able to do that because it’s printed in a very specific manner. If you’re able to figure out a way to get that to work, feel free to open a PR!